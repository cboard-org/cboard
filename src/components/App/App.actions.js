import {
  FINISH_FIRST_VISIT,
  UPDATE_DISPLAY_SETTINGS,
  UPDATE_NAVIGATION_SETTINGS
} from './App.constants';

export function updateDisplaySettings(payload = {}) {
  return {
    type: UPDATE_DISPLAY_SETTINGS,
    payload
  };
}

export function updateNavigationSettings(payload = {}) {
  return {
    type: UPDATE_NAVIGATION_SETTINGS,
    payload
  };
}

export function finishFirstVisit() {
  return {
    type: FINISH_FIRST_VISIT
  };
}
