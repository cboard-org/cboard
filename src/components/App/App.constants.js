// action types constants
export const FINISH_FIRST_VISIT = 'cboard/App/FINISH_FIRST_VISIT';
export const UPDATE_CONNECTIVITY = 'cboard/App/UPDATE_CONNECTIVITY';
export const UPDATE_DISPLAY_SETTINGS = 'cboard/App/UPDATE_DISPLAY_SETTINGS';
export const UPDATE_NAVIGATION_SETTINGS =
  'cboard/App/UPDATE_NAVIGATION_SETTINGS';

// language constants
export const DEFAULT_LANG = 'en-US';
export const APP_LANGS = [
  'ar-SA',
  'be-BY',
  'bn-BD',
  'cs-CZ',
  'da-DK',
  'de-DE',
  'el-GR',
  'en-US',
  'en-GB',
  'es-ES',
  'fi-FI',
  'fr-FR',
  'he-IL',
  'hi-IN',
  'hr-HR',
  'hu-HU',
  'id-ID',
  'it-IT',
  'ja-JP',
  'km-KH',
  'ko-KR',
  'ne-NP',
  'nl-NL',
  'no-NO',
  'pl-PL',
  'pt-BR',
  'pt-PT',
  'ro-RO',
  'ru-MD',
  'ru-RU',
  'si-LK',
  'sk-SK',
  'sr-SP',
  'sv-SE',
  'th-TH',
  'tr-TR',
  'uk-UA',
  'vi-VN',
  'zh-CN',
  'zu-ZA' // for crowdin contextual translation
];
