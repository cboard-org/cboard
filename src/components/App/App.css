.App {
  height: 100%;
}

.Cboard__FontSize__Standard {
  font-size: 16px;
}

.Cboard__FontSize__Large {
  font-size: 18px;
}

.Cboard__FontSize__ExtraLarge {
  font-size: 22px;
}
