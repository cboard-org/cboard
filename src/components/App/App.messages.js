import { defineMessages } from 'react-intl';

export default defineMessages({
  save: {
    id: 'cboard.containers.App.save',
    defaultMessage: 'Save'
  },
  newContentAvailable: {
    id: 'cboard.containers.App.newContentAvailable',
    defaultMessage: 'New content is available; please refresh.'
  },
  contentIsCached: {
    id: 'cboard.containers.App.contentIsCached',
    defaultMessage: 'Content is cached for offline use.'
  }
});
