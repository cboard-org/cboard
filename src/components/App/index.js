export { default } from './App.wrapper';
