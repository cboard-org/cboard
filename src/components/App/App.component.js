import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';

import registerServiceWorker from '../../registerServiceWorker';
import BoardWrapper from '../Board';
import Notifications from '../Notifications';
import './App.css';

export class App extends Component {
  static propTypes = {
    /**
     * App language
     */
    lang: PropTypes.string.isRequired,
    /**
     * App direction
     */
    dir: PropTypes.string.isRequired
  };

  componentDidMount() {
    const { onNewContentAvailable, onContentCached } = this.props;
    registerServiceWorker(onNewContentAvailable, onContentCached);
  }

  render() {
    const { lang, dir } = this.props;

    return (
      <div className="App">
        <Helmet>
          <html lang={lang} dir={dir} />
        </Helmet>
        <BoardWrapper />
        <Notifications />
      </div>
    );
  }
}

export default App;
