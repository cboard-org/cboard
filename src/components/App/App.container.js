import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { injectIntl, intlShape } from 'react-intl';

import registerServiceWorker from '../../registerServiceWorker';
import { showNotification } from '../Notifications/Notifications.actions';
import { isFirstVisit, isLogged } from './App.selectors';
import messages from './App.messages';
import App from './App.component';
import { DISPLAY_SIZE_STANDARD } from '../Settings/Display/Display.constants';

export class AppContainer extends Component {
  static propTypes = {
    /**
     * App language direction
     */
    dir: PropTypes.string.isRequired,
    /**
     * @ignore
     */
    intl: intlShape.isRequired,
    /**
     * If 'true', user first visit
     */
    isFirstVisit: PropTypes.bool,
    /**
     * If 'true', user is logged in
     */
    isLogged: PropTypes.bool,
    /**
     * App language
     */
    lang: PropTypes.string.isRequired,
    displaySettings: PropTypes.object.isRequired
  };

  componentDidMount() {
    registerServiceWorker(
      this.handleNewContentAvailable,
      this.handleContentCached
    );
  }

  handleNewContentAvailable = () => {
    const { intl, showNotification } = this.props;
    showNotification(intl.formatMessage(messages.newContentAvailable));
  };

  handleContentCached = () => {
    const { intl, showNotification } = this.props;
    showNotification(intl.formatMessage(messages.contentIsCached));
  };

  render() {
    const { dir, isFirstVisit, isLogged, lang, displaySettings } = this.props;

    const uiSize = displaySettings.uiSize || DISPLAY_SIZE_STANDARD;
    const fontSize = displaySettings.fontSize || DISPLAY_SIZE_STANDARD;
    const classes = [
      'Cboard__DisplaySettings',
      `Cboard__UISize__${uiSize}`,
      `Cboard__FontSize__${fontSize}`
    ];

    const htmlElement = document.getElementsByTagName('html')[0];
    htmlElement.className = classes.join(' ');

    return (
      <App
        dir={dir}
        isFirstVisit={isFirstVisit}
        isLogged={isLogged}
        lang={lang}
      />
    );
  }
}

const mapStateToProps = state => ({
  dir: state.language.dir,
  isFirstVisit: isFirstVisit(state),
  isLogged: isLogged(state),
  lang: state.language.lang,
  displaySettings: state.app.displaySettings
});

const mapDispatchToProps = {
  showNotification
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(injectIntl(AppContainer));
