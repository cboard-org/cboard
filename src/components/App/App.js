import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Helmet from 'react-helmet';
import { injectIntl, intlShape } from 'react-intl';

import registerServiceWorker from '../../registerServiceWorker';
import { showNotification } from '../Notifications/Notifications.actions';
import BoardContainer from '../Board';
import Notifications from '../Notifications';
// import WelcomeScreen from '../WelcomeScreen';
import messages from './App.messages';
import './App.css';

export class App extends Component {
  static propTypes = {
    /**
     * @ignore
     */
    intl: intlShape.isRequired,
    /**
     * App language
     */
    lang: PropTypes.string.isRequired,
    /**
     * App direction
     */
    dir: PropTypes.string.isRequired
  };

  componentDidMount() {
    const { intl, showNotification } = this.props;

    const onNewContentAvailable = () => {
      showNotification(intl.formatMessage(messages.newContentAvailable));
    };

    const onContentCached = () => {
      showNotification(intl.formatMessage(messages.contentIsCached));
    };

    registerServiceWorker(onNewContentAvailable, onContentCached);
  }

  render() {
    // const { dir, isFirstVisit, lang } = this.props;
    const { dir, lang } = this.props;

    return (
      <div className="App" onClick={() => {}}>
        <Helmet>
          <html lang={lang} dir={dir} />
        </Helmet>
        {/* {isFirstVisit ? <WelcomeScreen /> : <BoardContainer />} */}
        <BoardContainer />
        <Notifications />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  dir: state.language.dir,
  isFirstVisit: state.app.isFirstVisit,
  lang: state.language.lang
});

const mapDispatchToProps = {
  showNotification
};

export default connect(mapStateToProps, mapDispatchToProps)(injectIntl(App));
