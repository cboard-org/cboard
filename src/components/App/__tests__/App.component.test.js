import React from 'react';
import { shallow } from 'enzyme';

import App from '../App.component';

jest.mock('../App.messages', () => {
  return {
    save: {
      id: 'cboard.components.App.save',
      defaultMessage: 'Save'
    },
    newContentAvailable: {
      id: 'cboard.components.App.newContentAvailable',
      defaultMessage: 'New content is available; please refresh.'
    },
    contentIsCached: {
      id: 'cboard.components.App.contentIsCached',
      defaultMessage: 'Content is cached for offline use.'
    }
  };
});
jest.mock('../../Settings/Scanning/Scanning.messages', () => {
  return {
    scanning: {
      id: 'cboard.components.Settings.Scanning.scanning',
      defaultMessage: 'Scanning'
    },
    enable: {
      id: 'cboard.components.Settings.Scanning.enable',
      defaultMessage: 'Enable'
    },
    enableSecondary: {
      id: 'cboard.components.Settings.Scanning.enableSecondary',
      defaultMessage: 'Start scanning boards immediately'
    },
    delay: {
      id: 'cboard.components.Settings.Scanning.delay',
      defaultMessage: 'Time delay'
    },
    delaySecondary: {
      id: 'cboard.components.Settings.Scanning.delaySecondary',
      defaultMessage: 'Time between two consecutive scanning highlights'
    },
    method: {
      id: 'cboard.components.Settings.Scanning.method',
      defaultMessage: 'Scan method'
    },
    methodSecondary: {
      id: 'cboard.components.Settings.Scanning.methodSecondary',
      defaultMessage: 'Method to be used for board exploration'
    },
    seconds: {
      id: 'cboard.components.Settings.Scanning.seconds',
      defaultMessage: '{value} seconds'
    },
    automatic: {
      id: 'cboard.components.Settings.Scanning.automatic',
      defaultMessage: 'Automatic'
    },
    manual: {
      id: 'cboard.components.Settings.Scanning.manual',
      defaultMessage: 'Manual'
    }
  };
});

describe('App component tests', () => {
  const props = {
    dir: '/',
    isFirstVisit: true,
    isLogged: true,
    lang: 'es-ES',
    logout: () => {},
    user: {},
    messages: {}
  };

  it('renders without crashing', () => {
    shallow(<App {...props} />);
  });
  it('Matches Snapshot', () => {
    const wrapper = shallow(<App {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
