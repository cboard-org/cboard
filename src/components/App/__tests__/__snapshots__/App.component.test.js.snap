// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component tests Matches Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    dir="/"
    isFirstVisit={true}
    isLogged={true}
    lang="es-ES"
    logout={[Function]}
    messages={Object {}}
    user={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <HelmetWrapper
          defer={true}
          encodeSpecialCharacters={true}
        >
          <html
            dir="/"
            lang="es-ES"
          />
        </HelmetWrapper>,
        <UNDEFINED />,
        <Route
          component={
            Object {
              "mapDispatchToProps": [Function],
              "mapStateToProps": [Function],
              "mockDispatch": [MockFunction],
              "reactComponent": [Function],
            }
          }
          exact={true}
          path="/"
        />,
        <Switch>
          <RedirectIfLogged
            component={[Function]}
            isLogged={true}
            path="/login-signup"
            to="/"
          />
          <Route
            component={[Function]}
            path="/settings"
          />
          <Route
            component={[Function]}
            path="/activate/:url"
          />
          <Route
            component={
              Object {
                "mapDispatchToProps": [Function],
                "mapStateToProps": [Function],
                "mockDispatch": [MockFunction],
                "reactComponent": [Function],
              }
            }
            path="/login/:type/callback"
          />
          <Route
            component={
              Object {
                "mapDispatchToProps": [Function],
                "mapStateToProps": [Function],
                "mockDispatch": [MockFunction],
                "reactComponent": [Function],
              }
            }
            path="/board/:id"
          />
          <Route
            component={[Function]}
          />
        </Switch>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <html
            dir="/"
            lang="es-ES"
          />,
          "defer": true,
          "encodeSpecialCharacters": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "dir": "/",
            "lang": "es-ES",
          },
          "ref": null,
          "rendered": null,
          "type": "html",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": Object {
          "mapDispatchToProps": [Function],
          "mapStateToProps": [Function],
          "mockDispatch": [MockFunction],
          "reactComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": Object {
            "mapDispatchToProps": [Function],
            "mapStateToProps": [Function],
            "mockDispatch": [MockFunction],
            "reactComponent": [Function],
          },
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <RedirectIfLogged
              component={[Function]}
              isLogged={true}
              path="/login-signup"
              to="/"
            />,
            <Route
              component={[Function]}
              path="/settings"
            />,
            <Route
              component={[Function]}
              path="/activate/:url"
            />,
            <Route
              component={
                Object {
                  "mapDispatchToProps": [Function],
                  "mapStateToProps": [Function],
                  "mockDispatch": [MockFunction],
                  "reactComponent": [Function],
                }
              }
              path="/login/:type/callback"
            />,
            <Route
              component={
                Object {
                  "mapDispatchToProps": [Function],
                  "mapStateToProps": [Function],
                  "mockDispatch": [MockFunction],
                  "reactComponent": [Function],
                }
              }
              path="/board/:id"
            />,
            <Route
              component={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "component": [Function],
              "isLogged": true,
              "path": "/login-signup",
              "to": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/settings",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/activate/:url",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": Object {
                "mapDispatchToProps": [Function],
                "mapStateToProps": [Function],
                "mockDispatch": [MockFunction],
                "reactComponent": [Function],
              },
              "path": "/login/:type/callback",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": Object {
                "mapDispatchToProps": [Function],
                "mapStateToProps": [Function],
                "mockDispatch": [MockFunction],
                "reactComponent": [Function],
              },
              "path": "/board/:id",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HelmetWrapper
            defer={true}
            encodeSpecialCharacters={true}
          >
            <html
              dir="/"
              lang="es-ES"
            />
          </HelmetWrapper>,
          <UNDEFINED />,
          <Route
            component={
              Object {
                "mapDispatchToProps": [Function],
                "mapStateToProps": [Function],
                "mockDispatch": [MockFunction],
                "reactComponent": [Function],
              }
            }
            exact={true}
            path="/"
          />,
          <Switch>
            <RedirectIfLogged
              component={[Function]}
              isLogged={true}
              path="/login-signup"
              to="/"
            />
            <Route
              component={[Function]}
              path="/settings"
            />
            <Route
              component={[Function]}
              path="/activate/:url"
            />
            <Route
              component={
                Object {
                  "mapDispatchToProps": [Function],
                  "mapStateToProps": [Function],
                  "mockDispatch": [MockFunction],
                  "reactComponent": [Function],
                }
              }
              path="/login/:type/callback"
            />
            <Route
              component={
                Object {
                  "mapDispatchToProps": [Function],
                  "mapStateToProps": [Function],
                  "mockDispatch": [MockFunction],
                  "reactComponent": [Function],
                }
              }
              path="/board/:id"
            />
            <Route
              component={[Function]}
            />
          </Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <html
              dir="/"
              lang="es-ES"
            />,
            "defer": true,
            "encodeSpecialCharacters": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "dir": "/",
              "lang": "es-ES",
            },
            "ref": null,
            "rendered": null,
            "type": "html",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": Object {
            "mapDispatchToProps": [Function],
            "mapStateToProps": [Function],
            "mockDispatch": [MockFunction],
            "reactComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": Object {
              "mapDispatchToProps": [Function],
              "mapStateToProps": [Function],
              "mockDispatch": [MockFunction],
              "reactComponent": [Function],
            },
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <RedirectIfLogged
                component={[Function]}
                isLogged={true}
                path="/login-signup"
                to="/"
              />,
              <Route
                component={[Function]}
                path="/settings"
              />,
              <Route
                component={[Function]}
                path="/activate/:url"
              />,
              <Route
                component={
                  Object {
                    "mapDispatchToProps": [Function],
                    "mapStateToProps": [Function],
                    "mockDispatch": [MockFunction],
                    "reactComponent": [Function],
                  }
                }
                path="/login/:type/callback"
              />,
              <Route
                component={
                  Object {
                    "mapDispatchToProps": [Function],
                    "mapStateToProps": [Function],
                    "mockDispatch": [MockFunction],
                    "reactComponent": [Function],
                  }
                }
                path="/board/:id"
              />,
              <Route
                component={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "component": [Function],
                "isLogged": true,
                "path": "/login-signup",
                "to": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/settings",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/activate/:url",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": Object {
                  "mapDispatchToProps": [Function],
                  "mapStateToProps": [Function],
                  "mockDispatch": [MockFunction],
                  "reactComponent": [Function],
                },
                "path": "/login/:type/callback",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": Object {
                  "mapDispatchToProps": [Function],
                  "mapStateToProps": [Function],
                  "mockDispatch": [MockFunction],
                  "reactComponent": [Function],
                },
                "path": "/board/:id",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
